### PR TITLE
[Migration][DO NOT MERGE] Fix linting

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #1334
* #1340
* #1345
* #1343
* __->__ #1333
* #1332
* #1331
* #1330
* #1329
* #1328

Cutoff from 991f82bcc99706cd8ea6b7f2b70ccec75db2dbd6.
Changes after the above commit but before the cutoff date will be migrated in follow-up within this stack.

### CODE REVIEW

File: .lintrunner.toml [hunk 1]
```diff
@@ -46,6 +46,21 @@ exclude_patterns = [
...
+    'onnxscript/rewriter/function_rule.py',  # FIXME
+    'onnxscript/_legacy_ir/irbuilder.py',  # FIXME
+    'onnxscript/optimizer/fold_constants_v0.py',  # FIXME
+    'onnxscript/rewriter/pattern.py',  # FIXME
+    'onnxscript/rewriter/onnxruntime/transformers/multihead_attention.py',  # FIXME
+    'onnxscript/tools/function_unittest_producer.py',  # FIXME
+    'onnxscript/_legacy_ir/visitor.py',  # FIXME
+    'onnxscript/_legacy_ir/protobuilder.py',  # FIXME
+    'onnxscript/rewriter/onnxruntime/transformers/layernorm.py',  # FIXME
+    'onnxscript/ir/serde.py',  # FIXME
```
Comment (justinchuby) [inline .lintrunner.toml:63]: Is there new errors?

**Review by justinchuby — APPROVED** [review]

File: .lintrunner.toml (hunk 1 above)
Comment (BowenBao) [reply .lintrunner.toml:63]: some attr not defined error, because CI onnx is not 1.16

Comment (justinchuby) [reply .lintrunner.toml:63]: Ok. Strange it wasn’t caught last time

**Comment by BowenBao** [comment]
Close as squashed and merged into main.

**Pull request closed without merging**